tests/e2e/scenarios/02_upgrade.js

```diff
@@ -0,0 +1,136 @@
+  downloadFolder: 'modules/autoupgrade/download',
+};
+
+/*
+Go to login page
+Check PS version
+Upgrade
+Log out
+Check new version
+ */
+describe(`Upgrade PrestaShop from '${global.PS_VERSION}' to '${global.PS_VERSION_UPGRADE_TO}'`, async () => {
+  // before and after functions
+  before(async function () {
+    browserContext = await helper.createBrowserContext(this.browser);
+
+    page = await helper.newTab(browserContext);
+  });
+
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  afterEach(async function () {
+    if (this.currentTest.state === 'failed') {
+      await page.screenshot({path: `./screenshots/failed-step-${failedStepNumber}.png`, fullPage: true});
+      failedStepNumber += 1;
+    }
+  });
+
+  it('should go to login page', async () => {
+    await loginPage.goTo(page, global.BO.URL);
+
+    const pageTitle = await loginPage.getPageTitle(page);
+    await expect(pageTitle).to.contains(loginPage.pageTitle);
+  });
+
+  it('should check PS version', async () => {
+    const psVersion = await loginPage.getPrestashopVersion(page);
+    await expect(psVersion).to.contains(global.PS_VERSION);
+  });
+
+  it('should login into BO with default user', async () => {
+    await loginPage.login(page, global.BO.EMAIL, global.BO.PASSWD);
+    await dashboardPage.closeOnboardingModal(page);
+
+    const pageTitle = await dashboardPage.getPageTitle(page);
+    await expect(pageTitle).to.contains(dashboardPage.pageTitle);
+  });
+
+  it('should go to modules manager page', async () => {
+    await dashboardPage.goToSubMenu(
+      page,
+      dashboardPage.modulesParentLink,
+      dashboardPage.moduleManagerLink,
+    );
+
+    const pageTitle = await moduleManagerPage.getPageTitle(page);
+    await expect(pageTitle).to.contains(moduleManagerPage.pageTitle);
+  });
+
+  it('should go to module configuration page', async () => {
+    await moduleManagerPage.searchModule(page, moduleData.tag, moduleData.name);
+    await moduleManagerPage.goToConfigurationPage(page, moduleData.name);
+
+    const pageTitle = await upgradeModulePage.getPageTitle(page);
+    await expect(pageTitle).to.contains(upgradeModulePage.pageTitle);
+  });
+
+  it('should fill \'Expert mode\' form', async () => {
+    const textResult = await upgradeModulePage.fillExpertModeForm(
+      page,
+      'Local archive',
+      global.ZIP_NAME,
+      global.PS_VERSION_UPGRADE_TO,
+    );
+
+    await expect(textResult).to.contain(upgradeModulePage.configResultValidationMessage);
+  });
+
+  it('should put the shop under maintenance and check if the checklist is all green', async () => {
+    await upgradeModulePage.putShopUnderMaintenance(page);
+
+    for (let i = 1; i <= 10; i++) {
+      const textResult = await upgradeModulePage.getRowImageContent(page, i);
+      await expect(textResult).to.equal('ok');
+    }
+  });
+
+  it('should click on \'UPGRADE PRESTASHOP NOW\' and wait for the upgrade', async () => {
```

Comment:
I think you should also activate the "Switch to the default theme" when doing a major upgrade (1.6 -> 1.7)

Reply:
For the now, this does not support upgrades from 1.6, because the the prestashop_test_lib does not allow it .